File: 594. Longest Harmonious Subsequence/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findLHS = function(nums) {
  if (nums.length < 2) return 0;

  let count = new Map();
  let longest = 0;

  for (let num of nums) {
    count.set(num, count.get(num) ? count.get(num) + 1 : 1);

    if (count.has(num + 1)) {
      longest = Math.max(longest, count.get(num + 1) + count.get(num));
    }

    if (count.has(num - 1)) {
      longest = Math.max(longest, count.get(num - 1) + count.get(num));
    }
  }

  return longest;
};

File: 594. Longest Harmonious Subsequence/solution.go
func findLHS(nums []int) int {
	m := make(map[int]int, len(nums))
	for _, v := range nums {
		m[v]++
	}

	max := 0
	for i, c1 := range m {
		if c2, ok := m[i+1]; ok {
			t := c1 + c2
			if max < t {
				max = t
			}
		}
	}
	return max
}